src/main/controllers/ChangeLegalRepresentativeController.ts

```diff
@@ -0,0 +1,77 @@
+import { Response } from 'express';
+
+import { Form } from '../components/form';
+import { AppRequest } from '../definitions/appRequest';
+import { LEGAL_REPRESENTATIVE_CHANGE_OPTIONS, PageUrls, TranslationKeys } from '../definitions/constants';
+import { FormContent, FormFields } from '../definitions/form';
+import { AnyRecord } from '../definitions/util-types';
+import { getPageContent } from '../helpers/FormHelper';
+import { conditionalRedirect, getLanguageParam } from '../helpers/RouterHelpers';
+import { getLogger } from '../logger';
+import { isFieldFilledIn } from '../validators/validator';
+
+const logger = getLogger('RespondentApplicationDetailsController');
+
+export default class ChangeLegalRepresentativeController {
+  private readonly form: Form;
+  private readonly changeLegalRepresentativeContent: FormContent = {
+    fields: {
+      legalRep: {
+        id: 'legalRep',
+        type: 'radios',
+        label: (l: AnyRecord): string => l.h1,
+        labelHidden: false,
+        labelSize: 'xl',
+        classes: 'govuk-radios--inline',
+        values: [
+          {
+            label: (l: AnyRecord): string => l.legalRepresentativeChange,
+            value: LEGAL_REPRESENTATIVE_CHANGE_OPTIONS.change,
+          },
+          {
+            label: (l: AnyRecord): string => l.legalRepresentativeRemove,
+            value: LEGAL_REPRESENTATIVE_CHANGE_OPTIONS.remove,
+          },
+        ],
+        validator: isFieldFilledIn,
+      },
+    },
+    submit: {
+      text: (l: AnyRecord): string => l.submitBtn,
+    },
+  };
+
+  constructor() {
+    this.form = new Form(<FormFields>this.changeLegalRepresentativeContent.fields);
+  }
+
+  public post = async (req: AppRequest, res: Response): Promise<void> => {
+    try {
+      const redirectUrl = conditionalRedirect(
+        req,
+        this.form.getFormFields(),
+        LEGAL_REPRESENTATIVE_CHANGE_OPTIONS.change
+      )
+        ? PageUrls.APPOINT_LEGAL_REPRESENTATIVE + getLanguageParam(req.url)
+        : PageUrls.CONTACT_TRIBUNAL + PageUrls.CONTACT_TRIBUNAL + getLanguageParam(req.url);
```

Comment:
Do we need two "PageUrls.CONTACT_TRIBUNAL" here?

Reply:
Unfortunately yes Cindy because It is going twice to the controller to retrieve apply to tribunal page as in the image below. 
<img width="1918" height="820" alt="Screenshot 2025-08-15 at 11 54 28" src="https://github.com/user-attachments/assets/0ebbe4be-79c7-4bed-9851-b19212e2c2b0" />